crates/evm/src/evm/mod.rs

```diff
@@ -20,13 +20,13 @@ pub(crate) mod call;
 #[cfg(feature = "native")]
 pub(crate) mod error;
 
-#[cfg(test)]
+#[cfg(all(test, feature = "native"))]
```

Comment:
i see many of this cfg() changes.

I don't understand why some tests module is under native feature flag.

Reply:
Because they test mods that are behind "native" flag.

Reply:
hmm, 

what about listing the crate itself with native flag under `[dev-dependencies]`

Reply:
What for? We test with all features, so those tests will still be tested. That's a fix for `make check-features` only

Reply:
A proper way to test it would be to move tests into that module instead of keeping `test_MODULE.rs` modules.